README.md
###### [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/hellarafa/Kali-Setup-Script/issues) [![Hits](http://hits.dwyl.com/{hellarafa}/{Kali-Setup-Script}.svg)](http://hits.dwyl.com/{hellarafa}/{Kali-Setup-Script})
# Kali Boost | v1.1.2

Installs popular pentesting tools in the /opt folder. Just run the script and watch the magic happen. Best if used on a fresh Kali install. Do not eat. :100:

### Installation:
```bash
wget https://raw.githubusercontent.com/hellarafa/Kali-Boost/master/kali-boost.sh
chmod +x kali-boost.sh
./kali-boost.sh -s
```
##### One-liner, cause you know, we're all lazy...
```bash
curl -s https://raw.githubusercontent.com/hellarafa/Kali-Boost/master/kali-boost.sh | bash /dev/stdin -s
```


### Usage:
```
#########################################################

Usage: kali-boost.sh [-args]

    OPTIONS:
    -s     Start the installation.
    -r     Install ONLY the red team stuff.
    -e     Install EVERYTHING.
    -u     Update Kali.
    -v     Be verbose.
    -l     List all the tools that are part of this script.
    -b     Show the banner.
    -h     Show this help screen.

#########################################################
```



CHANGELOG.md
# Changelog
All future changes from Version 1.0.4 and onward will be documented in this file.
## [1.1.2b] - 2018-09-06
### Added
#### Tools
- crowbar from [@galkan](https://github.com/galkan/crowbar)

## [1.1.2a] - 2018-07-18
### Added
#### Tools
- onesixtyone from [@trailofbits](https://github.com/trailofbits/onesixtyone)

## [1.1.2] - 2018-05-23
### Added
#### Tools
- psgetsystem from [@decoder-it](https://github.com/decoder-it/psgetsystem)
- gdebi from [Michael Vogt](https://launchpad.net/gdebi)
- gobuster from [@OJ](https://github.com/OJ/gobuster)

### Changed
- Big change!! Renamed tool and repo from Kali-Setup-Script to Kali-Boost. Such a better name.
- Moved everything to a new repo. Just for the cleanliness.

## [1.1.1] - 2018-04-18
### Added
- Verbosity option. Had to rewrite some of the code but it works. Default is silent or no verbose output unless user invokes with -v.
- Function that lists all the tools that are going to be installed. Invoke with -l. Thats lowercase L as in lollipop or loser or legit.
- Update function. Literally just updates Kali for you. Invoke with -u.
#### Tools
- dirsearch from [@maurosoria](https://github.com/maurosoria/dirsearch)
-

---
## [1.1.0] - 2018-04-12
### Added
- Added comments like a legit programmer.
#### Tools
- wfuzz from [@xmendez](https://github.com/xmendez/wfuzz)
- Probable-Wordlists from [@berzerk0](https://github.com/berzerk0/Probable-Wordlists)
- LinEnum from [@cheetz](https://github.com/cheetz/LinEnum)
- MeterSSH from [@trustedsec](https://github.com/trustedsec/meterssh)
- The-Backdoor-Factory from [@secretsquirrel](https://github.com/secretsquirrel/the-backdoor-factory)
- SigThief from [@secretSquirrel](https://github.com/secretsquirrel/SigThief)
- Veil-Framework from [@Veil-Framework](https://github.com/Veil-Framework/Veil)

### Changed
- Completely changed the logical operation of the script. First, I added everything into its own function. Second, I added arguments to help users decide what to install. Lastly, I added a call table that calls each function and in turn, installs each tool.
- New version because of the huge amount of changes I made.
- Fixed the Empire installation. It still didn't work.
- Made everything 1337 colorful with a nice checkmark. 
- Changed output messages.
- Included a debug function as well as some internal functions to make the output look nicer.

---
## [1.0.4] - 2018-04-02
### Added
- This changelog
#### Tools
- Whonow DNS Server from [@brannondorsey](https://github.com/brannondorsey)
- Burp vulners plugin from [@vulnersCom](https://github.com/vulnersCom)
- dotdotslash from [@jcesarstef](https://github.com/jcesarstef)

### Changed
- The Empire section was not actually installing Empire if th:wqe user selected "Yes". Added a couple of lines to fix it.


rafa_setup_kali.sh
#!/bin/bash

#################### START
#
#  VERSION 1.1.0
#  This is where the program starts. I call the section that follows the call table. If you 
#  comment out a line, that program does not get installed. Simple as that. As a default, all
#  tools are installed except the REDTEAM EVIL STUFF. You can install it if you'd like.

start() {
    banner
    cme_bleeding_edge #1
    empire #2
    firefox_password_cracker #3
    eyewitness #4
    impacket_tools #5
    mimikittenz #6
    mimipenguin #7
    noSQLMap #8
    masscan #9
    powershellPopup #10
    powersploit #11
    firefox_password_cracker2 #12
    rtfm #13
    seclists #14
    wifite #15
    wifite2 #16
    nishang #17
    worawitms17 #18
    vulscan #19
    vulners #20
    credninja #21
    elfStrings #22
    dotdotslash #23
    burp_vulners #24
    whonow #25
    wfuzz #26
    probableWordlists #27

    #ADD
    #mimikatz
    #peterkim's tools
    #https://github.com/gentilkiwi/mimikatz/releases

    #REDTEAM SCRIPTS
#    redteam #uncomment this line to invoke. Have fun.
    echo -e $lightCyan"\n Finished installing all the tools! Have a great day."$reset.
}

redteam() {
    windowsHacks
}


cme_bleeding_edge() {
	echo -e $green"Installing: CME Bleeding-Edge"$reset
	    if [ ! -d "CrackMapExec" ]; then
            git clone https://github.com/byt3bl33d3r/CrackMapExec.git
            apt-get install -y libssl-dev libffi-dev python-dev build-essential
            ret="$?"
            (cd /opt/CrackMapExec && git submodule init && git submodule update --recursive)
            (cd /opt/CrackMapExec && python setup.py install)
            success "Successfully installed CrackMapExec Bleeding Edge"
	        debug
        else
            skipmsg "Found CrackMapExec Bleeding-Edge in /opt. Skipping installation."
	    fi
	sleep 1
}

empire() {
	echo -e $green"Installing: Empire"$reset
	    if [ -d "Empire" ]; then
            skipmsg "Found Empire in /opt. Skipping installation."
            read -p "Do you want to update instead (y/n)? " choice2
            case "$choice2" in
                [yY]|[yYeEsS] ) (cd /opt/Empire && git pull)
                            ret="$?"
                            success "Successfully updated Empire."
                            ;;
                [nN]|[nNoO] ) skipmsg "Not updating Empire.";;
                * ) skipmsg "Not updating Empire.";;
            esac
	    fi
	    if [ ! -d "Empire" ]; then
            git clone https://github.com/EmpireProject/Empire.git
            read -p "Empire requires extra installation steps. Do you want to continue with the Empire install? (y/n)? " choice1
            case "$choice1" in
                [yY]|[yYeEsS] ) (cd Empire && ./setup/install.sh)
                            ret="$?"
                            success "Successfully installed Empire."
                            ;;
                [nN]|[nNoO] ) skipmsg "Skipping installation. Manually run install.sh in the EmpireProject folder to install.";;
                * ) skipmsg "Skipping installation. Manually run install.sh in the EmpireProject folder to install.";;
            esac
	    fi
        debug
	sleep 1
}

firefox_password_cracker() {
	echo -e $green"Installing: ffpasscracker"$reset
	    if [ ! -d "ffpasscracker" ]; then
            git clone https://github.com/pradeep1288/ffpasscracker.git
            ret="$?"
            success "Successfully installed ffpasscracker."
            debug
        else
            skipmsg "Found ffpasscracker script in /opt. Skipping installation."
	    fi
	sleep 1
}

eyewitness() {
	echo -e $green"Installing: EyeWitness"$reset
	    if [ ! -d "EyeWitness" ]; then
            git clone https://github.com/ChrisTruncer/EyeWitness.git
            (cd /opt/EyeWitness/setup/ && ./setup.sh )
            ret="$?"
            success "Successfully installed EyeWitness."
            debug
        else
            skipmsg "Found EyeWitness in /opt. Skipping installation."
	    fi
	sleep 1
}

impacket_tools() {
	echo -e $green"Installing: Impacket"$reset
	    if [ ! -d "impacket" ]; then
            git clone https://github.com/CoreSecurity/impacket.git
            (cd impacket && python setup.py install)
            ret="$?"
            success "Successfully installed Impacket."
            debug
        else
            skipmsg "Found impacket in /opt. Skipping installation."
	    fi
	sleep 1
}

mimikittenz() {
	echo -e $green"Installing: mimikittenz"$reset
	    if [ ! -d "mimikittenz" ]; then
		    git clone https://github.com/putterpanda/mimikittenz.git
	        ret="$?"
            success "Successfully installed mimikittenz."
            debug
        else
		    skipmsg "Found mimikittenz in /opt. Skipping installation."
	    fi
	sleep 1
}

mimipenguin() {
	echo -e $green"Installing: mimipenguin"$reset
	    if [ ! -d "mimipenguin" ]; then
		    git clone https://github.com/huntergregal/mimipenguin.git
	        ret="$?"
            success "Successfully installed mimipenguin."
            debug
        else
		    skipmsg "Found mimipenguin in /opt. Skipping installation."
	    fi
	sleep 1
}

noSQLMap() {
	echo -e $green"Installing: NoSQLMap"$reset
	    if [ ! -d "NoSQLMap" ]; then
		    git clone https://github.com/codingo/NoSQLMap.git
	        ret="$?"
            success "Successfully installed NoSQLMap."
            debug
        else
	        skipmsg "Found mimipenguin in /opt. Skipping installation."
	    fi
	sleep 1
}

masscan() {
	echo -e $green"Installing: Masscan"$reset
	    if [ ! -d "masscan" ]; then
            git clone https://github.com/robertdavidgraham/masscan.git
            apt-get install -y git gcc make libpcap-dev
            (cd /opt/masscan && make -j && make install)
	        ret="$?"
            success "Successfully installed Masscan."
            debug
        else
            skipmsg "Found Masscan in /opt. Skipping installation."
	    fi
	sleep 1
}

powershellPopup() {
	echo -e $green"Installing: PowerShell_Popup"$reset
	    if [ ! -d "PowerShell_Popup" ]; then
	        git clone https://github.com/cheetz/PowerShell_Popup.git
	        ret="$?"
            success "Successfully installed PowerShell_Popup."
            debug
        else
	        skipmsg "Found PowerShell Popup in /opt. Skipping installation."
	    fi
	sleep 1
}

powersploit() {
	echo -e $green"Installing: PowerSploit"$reset
	    if [ ! -d "PowerSploit" ]; then
	        git clone https://github.com/PowerShellMafia/PowerSploit.git
	        ret="$?"
            success "Successfully installed PowerSploit."
            debug
        else
	        skipmsg "Found PowerSploit in /opt. Skipping installation."
	    fi
	sleep 1
}

firefox_password_cracker2() {
	echo -e $green"Installing: python-ffpassdecrypt"$reset
	    if [ ! -d "python-ffpassdecrypt" ]; then
            git clone https://github.com/nyov/python-ffpassdecrypt.git
            ret="$?"
            success "Successfully installed python-ffpassdecrypt."
            debug
	    else
            skipmsg "Found python-ffpassdecrypt in /opt. Skipping installation."
	    fi
	sleep 1
}

rtfm() {
	echo -e $green"Installing: rtfm"$reset
	    if [ ! -d "rtfm" ]; then
	        git clone https://github.com/leostat/rtfm.git
            ret="$?"
            success "Successfully installed rtfm."
            debug
	        chmod +x /opt/rtfm/rtfm.py
	    else
	        skipmsg "Found rftm in /opt. Skipping installation."
	    fi
	sleep 1
}

seclists() {
	echo -e $green"Installing: SecLists"$reset
	    if [ ! -d "SecLists" ]; then
	        git clone https://github.com/danielmiessler/SecLists.git
	        ret="$?"
            success "Successfully installed SecLists."
            debug
        else
	        skipmsg "Found Seclists in /opt. Skipping installation."
	    fi
	sleep 1
}

wifite() {
	echo -e $green"Installing: wifite"$reset
	    if [ ! -d "wifite" ]; then
	        git clone https://github.com/derv82/wifite.git
	        ret="$?"
            success "Successfully installed wifite."
            debug
        else
	        skipmsg "Found wifite in /opt. Skipping installation."
	    fi
	sleep 1
}

wifite2() {
	echo -e $green"Installing: wifite2"$reset
	    if [ ! -d "wifite2" ]; then
	        git clone https://github.com/derv82/wifite2.git
	        ret="$?"
            success "Successfully installed wifite2."
            debug
        else
	        skipmsg "Found wifite2 in /opt. Skipping installation."
	    fi
	sleep 1
}

nishang() {
	echo -e $green"Installing: nishang"$reset
	    if [ ! -d "./nishang/" ]; then
	        git clone https://github.com/samratashok/nishang.git 
	        ret="$?"
            success "Successfully installed nishang."
            debug
        else
	        skipmsg "Found nishang in /opt. Skipping installation."
	    fi
	sleep 1
}

worawitms17() {
	echo -e $green"Installing: Worawit's MS17-010"$reset
	    if [ ! -d "MS17-010" ]; then
            git clone https://github.com/worawit/MS17-010.git
	        ret="$?"
            success "Successfully installed Worawit's MS17-010."
            debug
        else
	        skipmsg "Found MS17-010 in /opt. Skipping installation."
	    fi
	sleep 1
}

vulscan() {
	echo -e $green"Installing: Vulscan"$reset
	    if [ ! -d "vulscan" ]; then
            git clone https://github.com/scipag/vulscan.git
	        ret="$?"
            cp -r vulscan/ /usr/share/nmap/scripts/
	        success "Successfully installed Vulscan. Find it in your nmap scripts."
            debug
        else
	        skipmsg "Found vulscan in /opt. Skipping installation."
	    fi
	sleep 1
}

vulners() {
	echo -e $green"Installing: Vulners"$reset
	    if [ ! -d "nmap-vulners" ]; then
            git clone https://github.com/vulnersCom/nmap-vulners.git
            ret="$?"
	        cp nmap-vulners/vulners.nse /usr/share/nmap/scripts/
	        success "Successfully installed Vulners. Find it in your nmap scripts."
            debug
        else
	        skipmsg "Found Vulners Nmap Script in /opt. Skipping installation."
	    fi
	sleep 1
}

credninja() {
	echo -e $green"Installing: CredNinja"$reset
	    if [ ! -d "CredNinja" ]; then
            git clone https://github.com/Raikia/CredNinja.git
	        ret="$?"
            success "Successfully installed CredNinja."
            debug
        else
	        skipmsg "Found CredNinja in /opt. Skipping installation."
	    fi
	sleep 1
}

elfStrings() {
	echo -e $green"Installing: ELF-Strings"$reset
	    if [ ! -d "elf-strings" ]; then
	        git clone https://github.com/LloydLabs/elf-strings
            ret="$?"
            success "Successfully installed ELF-Strings."
            debug
	    else
	        skipmsg "Found ELF-Strings in /opt. Skipping installation."
	    fi
	sleep 1
}

dotdotslash() {
	echo -e $green"Installing: dotdotslash"$reset
	    if [ ! -d "dotdotslash" ]; then
	        git clone https://github.com/jcesarstef/dotdotslash.git
	        ret="$?"
            success "Successfully installed dotdotslash."
            debug
        else
            skipmsg "Found dotdotslash in /opt. Skipping installation."
	    fi
	sleep 1
}

burp_vulners() {
	echo -e $green"Installing: burp-vulners-scanner"$reset
	    if [ ! -d "burp-vulners-scanner" ]; then
	        git clone https://github.com/vulnersCom/burp-vulners-scanner.git
            ret="$?"
            success "Successfully downloaded burp-vulners-scanner."
	        echo -e $lightYellow"Don't forget to manually install this module in Burp."$reset
	        sleep 3
	    else
	        skipmsg "Found burp-vulners-scanner in /opt. Skipping installation."
	    fi
	sleep 1
}

whonow() {
	echo -e $green"Installing: whonow"$reset
	    if [ ! -d "whonow" ]; then
	        git clone https://github.com/brannondorsey/whonow.git
	        ret="$?"
            success "Successfully installed whonow."
            debug
        else
	        skipmsg "Found whonow in /opt. Skipping installation."
	    fi
	sleep 1
}

wfuzz() {
	echo -e $green"Installing: wfuzz"$reset
	    if [ ! -d "wfuzz" ]; then
	        git clone https://github.com/xmendez/wfuzz.git
	        ret="$?"
            success "Successfully installed wfuzz."
            debug
        else
	        skipmsg "Found wfuzz in /opt. Skipping installation."
	    fi
	sleep 1
}

probableWordlists(){
    echo -e $green"Installing: Probable-Wordlists"$reset
        if [ ! -d "Probable-Wordlists" ]; then
            git clone https://github.com/berzerk0/Probable-Wordlists.git
            ret="$?"
            success "Successfully installed Probable-Wordlists."
            debug
        else
            skipmsg "Found Probable-Wordlists in /opt. Skipping installation."
        fi
    sleep 1
}

#################### REDTEAM EVIL STUFF
#  These scripts are malicious in nature and are usually not used
#  in Enterprise/Business engagements. Therefore, the function that
#  calls these is commented out but feel free to install them if you'd
#  like. CCDC people, feel free to use and/or contribute. ;-)

windowsHacks() {
    echo -e $green"Installing: Windows-Hacks"$reset
        if [ ! -d "Windows-Hacks" ]; then
            git clone https://github.com/LazoCoder/Windows-Hacks.git
            ret="$?"
            success "Successfully installed Windows-Hacks."
            debug
        else
            skipmsg "Found Windows-Hacks in /opt. Skipping installation."
        fi
    sleep 1
}

#################### TEMPLATE
#  if you want to add your own tool, just add this template code around
#  a function then add it to the call table below and you'll be good 
#  to go. Have fun.

#echo -e $green"Installing: ___"$reset
#    if [ ! -d "___" ]; then
#        git clone https://github.com/___.git
#        ret="$?"
#        success "DANK AF FAM"
#        debug
#    else
#        skipmsg "Found ___ in /opt. Skipping installation."
#    fi
#sleep 1

banner() {
    # mandatory banner lol.
    echo -e $green".##....##....###....##.......####"$reset
    echo -e $green".##...##....##.##...##........##."$reset
    echo -e $green".##..##....##...##..##........##."$reset
    echo -e $green".#####....##.....##.##........##."$reset
    echo -e $green".##..##...#########.##........##."$reset
    echo -e $green".##...##..##.....##.##........##."$reset
    echo -e $green".##....##.##.....##.########.####"$reset
    echo -e $green"..######..########.########.##.....##.########."$reset
    echo -e $green".##....##.##..........##....##.....##.##.....##"$reset
    echo -e $green".##.......##..........##....##.....##.##.....##"$reset
    echo -e $green"..######..######......##....##.....##.########."$reset
    echo -e $green".......##.##..........##....##.....##.##......."$reset
    echo -e $green".##....##.##..........##....##.....##.##......."$reset
    echo -e $green"..######..########....##.....#######..##......."$reset
    echo -e $green"..######...######..########..####.########..########"$reset
    echo -e $green".##....##.##....##.##.....##..##..##.....##....##..."$reset
    echo -e $green".##.......##.......##.....##..##..##.....##....##..."$reset
    echo -e $green"..######..##.......########...##..########.....##..."$reset
    echo -e $green".......##.##.......##...##....##..##...........##..."$reset
    echo -e $green".##....##.##....##.##....##...##..##...........##..."$reset
    echo -e $green"..######...######..##.....##.####.##...........##..."$reset
    echo -e "=============================="
    echo -e "===$lightCyan Version:$reset$lightYellow 1.1.0         $reset==="
    echo -e "===$lightCyan Last Updated:$reset$lightYellow 04-05-18 $reset==="
    echo -e "===$lightCyan Made by @hellarafa <3  $reset==="
    echo -e "=============================="
    echo ""
}

#################### PROGRAM FUNCTIONS
# Functions that make the program run nicer.
#

msg() {
    printf '%b\n\n' "$1" >&2
}

success() {
    if [ "$ret" -eq '0' ]; then
        msg "\33[32m[✔]\33[0m ${1}${2}"
    fi
}

error() {
    msg "\33[31m[✘]\33[0m ${1}${2}"
    exit 1
}

skipmsg(){
    msg "\33[31m[✘]\33[0m ${1}${2}"
}

debug() {
    if [ "$debug_mode" -eq '1'  ] && [ "$ret" -gt '1'  ]; then
        msg "An error occurred in function \"${FUNCNAME[$i+1]}\" on line ${BASH_LINENO[$i+1]}, I'm sorry for that."
    fi
}

#################### SETUP PARAMETERS
# Just all the setup parameters, variables and stuff.
#
debug_mode='0'
readonly reset="\e[0m"
readonly red="\e[31m"
readonly green="\e[38;5;46m"
readonly lightYellow="\e[93m"
readonly lightCyan="\e[96m"

cd /opt
#start
banner
